activerecord/lib/active_record/railties/databases.rake

```diff
@@ -40,9 +40,14 @@ db_namespace = namespace :db do
     end
   end
 
-  desc "Creates the database from DATABASE_URL or config/database.yml for the current RAILS_ENV (use db:create:all to create all databases in the config). Without RAILS_ENV or when RAILS_ENV is development, it defaults to creating the development and test databases, except when DATABASE_URL is present."
+  desc "Creates the database for DATABASE_NAME or from DATABASE_URL or config/database.yml for the current RAILS_ENV (use db:create:all to create all databases in the config). Without RAILS_ENV or when RAILS_ENV is development, it defaults to creating the development and test databases, except when DATABASE_URL is present."
   task create: [:load_config] do
-    ActiveRecord::Tasks::DatabaseTasks.create_current
+    if name = ENV["DATABASE_NAME"]
+      db_config = ActiveRecord::Base.configurations.configs_for(env_name: Rails.env, name: name)
+      ActiveRecord::Tasks::DatabaseTasks.create(db_config)
+    else
+      ActiveRecord::Tasks::DatabaseTasks.create_current
```

Comment:
I think that for at least create and drop the `DATABASE_NAME` handling could be moved into `create_current` and `drop_current` to reduce the complexity of the rake task code.